app/javascript/utils/form/validations.js

```diff
@@ -125,4 +125,8 @@ validate.isValidDate = (date, errors, options = {}) => {
   return errors
 }
 
+export const touchAllFields = (form, touchedState) => {
```

Comment:
I'm not super clear on what touchedState is here

Reply:
`touchedState` is the current touched `state` from the form of all fields, so it is just an object with all of the field names and a boolean value, i.e. 

```js
{
  name: false,
  username: true,
  password: false
}
```

final-form keeps track of these at all times so on our validation calls we can simply grab that object from the form props and loop through the names and blur them ourselves with `form.blur`